test/e2e/tests/metric.go

```diff
@@ -96,6 +96,54 @@ var MetricTest = suite.ConformanceTest{
 				t.Errorf("failed to scrape metrics: %v", err)
 			}
 		})
+
+		t.Run("workqueue and restclient metrics are exposed", func(t *testing.T) {
+			expectedResponse := httputils.ExpectedResponse{
+				Request: httputils.Request{
+					Path: "/prom",
+				},
+				Response: httputils.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			// make sure listener is ready
+			httputils.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, expectedResponse)
+
+			// Check that workqueue and restclient metrics are registered
+			if err := wait.PollUntilContextTimeout(context.TODO(), time.Second, time.Minute, true,
+				func(_ context.Context) (done bool, err error) {
+					url, err := RetrieveURL(suite.Client, types.NamespacedName{
+						Namespace: "envoy-gateway-system",
+						Name:      "same-namespace-gw-metrics",
+					}, 19001, "/stats/prometheus")
```

Comment:
isn't this's a controller metric? looks like you're trying to get metrics from dataplane?

Reply:
Yup, you are right. I couldn't run e2e tests locally, so I decided to run them in ci). It should work now, could you rerun tests?

Reply:
@zirain fix linter